src/detection/os/os_windows.cpp

```diff
@@ -33,13 +33,21 @@ void ffDetectOSImpl(FFOSResult* os, const FFinstance* instance)
         return;
     }
 
-    ffGetWmiObjString(pclsObj, L"Caption", &os->variant); // Microsoft Windows 11 家庭中文版
+    ffGetWmiObjString(pclsObj, L"Caption", &os->variant); // Microsoft Windows Server 2022 Datacenter / Microsoft Windows 11 家庭中文版
     if(ffStrbufStartsWithS(&os->variant, "Microsoft Windows "))
     {
         ffStrbufAppendS(&os->name, "Microsoft Windows");
         ffStrbufAppendS(&os->prettyName, "Windows");
 
-        ffStrbufSubstrAfter(&os->variant, strlen("Microsoft Windows ") - 1); // 11 家庭中文版
+        ffStrbufSubstrAfter(&os->variant, strlen("Microsoft Windows ") - 1); // Server 2022 Datacenter / 11 家庭中文版
+
+        if(ffStrbufStartsWithS(&os->variant, "Server "))
+        {
+            ffStrbufAppendS(&os->name, " Server");
+            ffStrbufAppendS(&os->prettyName, " Server");
+            ffStrbufSubstrAfter(&os->variant, strlen(" Server") - 1);
+        } // 2022 Datacenter / 11 家庭中文版
```

Comment:
Can the please keep everything in english?

Reply:
It was actually the Windows version printed by OS. I left it here to make the string operations easier to understand.

I removed the comments.